### ISSUE TITLE
docs(ic-id-protocol): specify how to decode an identity-provider-provided Bearer access_token, and use the decoded data to make authenticated ic-ref requests

### ISSUE THREAD

**Issue by gobengo** [dfinity/agent-js#135]
ic-id-protocol AuthenticationResponse#access_tokens are Bearer tokens. This is to minimize dev-learning-requirements (for webdevs who have done OAuth2/OIDC/SAML) about the IC, abstraction to not preclude back-compatable future changes to this IDP to support future extended methods of ic-ref Authentication.

However, the RP who received an AuthenticationResponse Bearer access_token must be able to use the token to make authenticated ic-ref requests.
* The current implementation of that is here:
  * createBearerToken: https://github.com/dfinity/agent-js/blob/identity-provider/2021-01-04/packages/identity-provider/src/protocol/ic-id-protocol.ts#L140
    * (this used to be something from a @dfinity/authentication prototype.toJSON, but the return type of that JS method changing shouldn't imply a change to this protocol that should always be backward-compatible with itself)
  * parseBearerToken: https://github.com/dfinity/agent-js/blob/identity-provider/2021-01-04/packages/identity-provider/src/protocol/ic-id-protocol.ts#L120

* [ ] Document how to parseBearerToken in English today
* [ ] Explain the results of parsing a bearer token, and link to documentation explaining why they're useful.
* [ ] Maybe amend this token to not be `AuthenticationResponse#token_type`='Bearer', and instead define a new `token_type` string and specify that it will be of this form.
  * This affords for evolving one or more token_types/versions over time, which is probably likely to be desired

Unblocks:
* proposals to change how this works today, or to specify alternate token_type or [response_type](https://tools.ietf.org/html/rfc6749#section-8.4) as OAuth2 extensions.
* internal and external devs reading English about how this works, not just code. This should make it possible for teammates and community members to someday innovate on Identity Providers in other languages/tradeoffs, etc.

**Comment by krpeacock** [comment]
I liked this pattern, but no longer relevant in agent-js
